src/sonic-yang-models/yang-models/sonic-buffer-port-ingress-profile-list.yang

```diff
@@ -0,0 +1,55 @@
+module sonic-buffer-port-ingress-profile-list {
+    namespace "http://github.com/Azure/sonic-buffer-port-ingress-profile-list";
+    prefix bpg;
+
+    import sonic-extension {
+        prefix sonic-ext;
+    }
+
+    import sonic-port {
+        prefix prt;
+    }
+
+    import sonic-buffer-profile {
+        prefix bpf;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC BUFFER PORT INGRESS PROFILE LIST";
+
+    revision 2021-07-01 {
+        description
+            "Initial revision.";
+    }
+
+
+    container sonic-buffer-port-ingress-profile-list {
+
+        container BUFFER_PORT_INGRESS_PROFILE_LIST {
+            list BUFFER_PORT_INGRESS_PROFILE_LIST_LIST {
+
+                key "port";
+
+                leaf port {
+                    type leafref {
+                        path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:name";
+                    }
+                }
+
+                leaf-list profile_list {
+                    type leafref {
+                        path "/bpf:sonic-buffer-profile/bpf:BUFFER_PROFILE/bpf:BUFFER_PROFILE_LIST/bpf:name";
```

Comment:
The same as egress profile list.

Reply:
No I dont think it can be checked for leafref.